### PR TITLE
[BE] Remove dependency on `six` and `future`

### PR DESCRIPTION
**Pull request by XuehaiPan**
Remove the Python 2 and 3 compatibility library [six](https://pypi.org/project/six) and [future](https://pypi.org/project/future) and `torch._six`. We only support Python 3.8+ now. It's time to retire them.

cc @mlazos @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @desertfire

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -66,10 +66,6 @@
     ignore = dirty
     path = third_party/python-peachpy
     url = https://github.com/malfet/PeachPy.git
-[submodule "third_party/python-six"]
```
Comment (Skylion007) [inline .gitmodules:69]: I actually don't think this is our dependency, but a dependency of one of the other third_party libs so it make break it. Let's see if it does though.

Comment (XuehaiPan) [reply .gitmodules]: It's a dependency of `cmake/External/nnpack.cmake`:

https://github.com/pytorch/pytorch/blob/a8fdfb4ba8a804c67d744a763fd9fa1f72d28590/cmake/External/nnpack.cmake#L48

re-added it.

**Review by malfet — APPROVED** [review]

File: .gitmodules (hunk 1 above)
Comment (malfet) [reply .gitmodules]: Hmm, we can always submit a PR to nnpack to fix the dependency on their end

**Review by Skylion007 — APPROVED** [review]

**Comment by wonjoolee95** [comment]
@XuehaiPan, can you update the XLA CI pin to `9b2fa918f3332dfc9e09d288a4aad8d9afd16f34`? Once CIs are green on both sides, we can coordinate a merge.

**Comment by Skylion007** [comment]
You still need to update the XLA pin

**Comment by weiwangmeta** [comment]
BC-breaking note

torch._six private API was removed and therefore this API call would error out with v2.0.0. Please follow instructions below to resolve breakages due to this. 

Example: 
Version 1.13.1:
`from torch._six import string_classes`
Version 2.0.0: 
`delete this import and replace string_classes with str
`

Example: 
Version 1.13.1: 
`from torch._six import int_classes`
Version 2.0.0: 
`delete this import and replace int_classes with int`

Example: 
Version 1.13.1: 
`from torch._six import inf, nan`
Version 2.0.0: 
`from torch import inf, nan`

Example: 
Version 1.13.1: 
`torch._six.string_classes was directly used`
Version 2.0.0: 
`replace  torch._six.string_classes  with "str"`

**Comment by albanD** [comment]
We should revisit all changes from `string_classes` to `str` to make sure to properly include bytes when relevant.
There are already 2 forward fixes for it: https://github.com/pytorch/pytorch/pull/97737 and https://github.com/pytorch/pytorch/pull/97789 but we should look through this diff and fix all the other ones pre-emptively!

@XuehaiPan would you have some time to look into that by any chance?

**Comment by XuehaiPan** [comment]
@albanD I opened a PR to re-add relevant `isinstance` checks for `bytes`.

- #97863

**Pull request closed without merging**